app/controllers/accounts_controller.rb

```diff
@@ -0,0 +1,44 @@
+class AccountsController < ApplicationController
+  include ActionController::HttpAuthentication::Basic::ControllerMethods
+
+  http_basic_authenticate_with(
+    name: Rails.application.credentials.http_basic_authenticate.user,
+    password: Rails.application.credentials.http_basic_authenticate.password,
+    only: :create
+  )
+
+  def create
+    if (account = Account.create_with_open_id_provider(params['sub'], params['provider'])).invalid?
+      return render_validation_errored account
+    end
+
+    render_created account
+  end
+
+  private
+
+  def render_created(account)
+    render json: {
+      id: account.id,
+      openIdProviders: [
+        {
+          sub: account.open_id_providers[0].sub,
```

Comment:
```suggestion
          sub: account.open_id_providers.first.sub,
```
これは趣味かもしれませんが、ruby では配列の [0] を取り出したい時には .first の方がよく使うような気がしています。

一応根拠のようなものを載せますが、「なぜその方がよいか」の説明はされていません。
ただよく見られていそうなリポジトリというだけです。

英語(おそらく原典):
https://github.com/rubocop/ruby-style-guide/blob/master/README.adoc#first-and-last
日本語:
https://github.com/fortissimo1997/ruby-style-guide/blob/japanese/README.ja.md#first-and-last

Reply:
そうですね、rubocopのスタイルに従って書いた方が良いというのが一般的とされていてrubocopでも

> 配列の最初の要素には#first、最後の要素には#lastでアクセスするのが望ましい

と記載があるのでその方が良いと思います。

https://techracho.bpsinc.jp/hachi8833/2017_05_15/38869